### PR TITLE
test(scale): chaos test for basic nexmark

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR introduces chaos tests for basic nexmark queries (q0 - q9). Random fragments will be picked and reschedule (scale) plans will be generated for them during the tests.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- #5655

### CODE REVIEW

File: src/tests/simulation_scale/src/ctl_ext.rs [hunk 1]
```diff
@@ -68,31 +79,33 @@ pub mod predicate {
...
-    pub fn can_scale() -> BoxedPredicate {
+    /// The fragment is able to be rescheduled. Used for locating random fragment.
+    pub fn can_reschedule() -> BoxedPredicate {
         let p = |f: &ProstFragment| {
-            // TODO: singleton fragments are also able to be migrated, may add a `can_migrate`
-            // predicate.
-            let distributed = f.distribution_type() == FragmentDistributionType::Hash;
+            // FIXME: we should already support reschedule source fragment, but there might be a bug
+            // of nexmark generator recovery.
+            let has_source = identity_contains("StreamSource")(f);
```
Comment (BugenZhao) [inline src/tests/simulation_scale/src/ctl_ext.rs:87]: If we reschedule the nexmark source executors, it seems the result will possibly be wrong. 🤔 cc @shanicky

Comment (BugenZhao) [reply src/tests/simulation_scale/src/ctl_ext.rs:87]: I'll take a look after this PR first.

**Comment by lmatz** [comment]
Is it possible to run this in daily tests or as a longevity test without much change? Thx!

Ideally, the daily test or longevity test runs for a much longer period of time,
and we can introduce much more chaos into this long period of time.

Aka, test time and level of chaos can be configured in an easy and non-hardcoded way.

**Comment by BugenZhao** [comment]
@lmatz

Sure. I've added the step to `main` and `main-cron` as well and configured a large test num for the cronjob.

For "long period of time", actually there should be some minor tweaks for different queries to make sure we've covered the operator execution for sufficient time. I'll refine this as more tests are added.

**Review by wangrunji0408 — APPROVED** [review]
Generally LGTM.

We can consider introducing a kind of script language to replace the hard-coding SQLs. 🥵

**Review by shanicky — APPROVED** [review]
LSSSSTM! 

The problem with nexmark source may be related to the spawned thread that generates the data separately, I'll figure out it later

**Pull request merged**